internal/service/advancedclustertpf/resource.go

```diff
@@ -55,40 +67,24 @@ func (r *rs) Schema(ctx context.Context, req resource.SchemaRequest, resp *resou
 func (r *rs) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
 	var plan TFModel
 	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
-	if resp.Diagnostics.HasError() {
-		return
-	}
-	sdkReq := NewAtlasReq(ctx, &plan, &resp.Diagnostics)
-	if resp.Diagnostics.HasError() {
-		return
-	}
-	err := StoreCreatePayload(sdkReq)
-	if err != nil {
-		resp.Diagnostics.AddError("errorCreate", fmt.Sprintf(errorCreate, err.Error()))
+	diags := &resp.Diagnostics
+	if diags.HasError() {
 		return
 	}
-	tfNewModel, shouldReturn := mockedSDK(ctx, &resp.Diagnostics, plan.Timeouts)
-	if shouldReturn {
-		return
+	summary, detail := r.createCluster(ctx, &plan, diags, resp)
```

Comment:
why not createCluster adds errors to diags (passing pointer) and doesn't need to return anything, and we check like previously: if diags.HasError() {...

Reply:
The problem is that you cannot do 
```go
return diags.AddError(summary, detail)
```
You will have to do it on two lines:
```go
diags.AddError(summary, detail)
return
```

As these functions have ~ 8 different return positions, I thought it was more readable to return summary, detail in one line.

WDYT?

Reply:
i see your point but I would go for removing  summary and detail from the response. 

createCluster can add logs, e.g. from the call to NewAtlasReq. 

I think it's confusing and can be error prone that we return empty summary and detail but we might have filled diags so there is actually an error. There are different ways to raise an error.

Also it's more "elegant" as main operations will look similar:

```
func (r *rs) Create(ctx context.Context, req resource.CreateRequest, resp *resource.CreateResponse) {
	var plan TFModel
	resp.Diagnostics.Append(req.Plan.Get(ctx, &plan)...)
	diags := &resp.Diagnostics
	if diags.HasError() {
		return
	}
        r.createCluster(ctx, &plan, diags, resp)
}
```

Reply:
I tried to make the change, and I agree it looks better and is more error safe.
3d3ebe5e2f42206b4d660685807e703761f05a22

Reply:
nice